controller/volume_controller.go

```diff
@@ -1015,6 +1019,52 @@ func (vc *VolumeController) getAutoBalancedReplicasSetting(v *longhorn.Volume) (
 	return setting, errors.Wrapf(err, "replica auto-balance is disabled")
 }
 
+func (vc *VolumeController) getUnmapMarkSnapChainRemovedSetting(v *longhorn.Volume) (bool, error) {
+	if v.Spec.UnmapMarkSnapChainRemoved == longhorn.UnmapMarkSnapChainRemovedEnabled {
+		return true, nil
+	}
+	if v.Spec.UnmapMarkSnapChainRemoved == longhorn.UnmapMarkSnapChainRemovedDisabled {
+		return false, nil
+	}
+
+	globalSetting, err := vc.ds.GetSetting(types.SettingNameUnmapMarkSnapChainRemoved)
+	if err != nil {
+		return false, err
+	}
+	switch globalSetting.Value {
+	case string(longhorn.UnmapMarkSnapChainRemovedEnabled):
+		return true, nil
+	case string(longhorn.UnmapMarkSnapChainRemovedDisabled):
+		return false, nil
+	default:
+		return false, fmt.Errorf("unknown value %v for setting %v", globalSetting.Value, types.SettingNameUnmapMarkSnapChainRemoved)
+	}
+}
+
+func (vc *VolumeController) syncVolumeUnmapMarkSnapChainRemovedSetting(v *longhorn.Volume, es map[string]*longhorn.Engine, rs map[string]*longhorn.Replica) error {
+	if es == nil && rs == nil {
+		return nil
+	}
+
+	unmapMarkEnabled, err := vc.getUnmapMarkSnapChainRemovedSetting(v)
+	if err != nil {
+		return err
+	}
+
+	if es != nil {
+		for _, e := range es {
+			e.Spec.UnmapMarkSnapChainRemovedEnabled = unmapMarkEnabled
```

Comment:
Break after this?

Reply:
We will blindly update this field for all engines and replicas.